modules/data.remote/inst/appeears2pecan.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+appears2pecan downloads remote sensing data using the AppEEARS API
+
+AppEEARS API documentation: https://lpdaacsvc.cr.usgs.gov/appeears/api/?language=Python%203#introduction
+
+Requires Python3
+
+Author(s): Ayush Prasad
+"""
+
+import requests as r
+import geopandas as gpd
+import getpass
+import time
+import os
+import cgi
+import json
+from gee_utils import get_sitename
+from datetime import datetime
+
+
+def appeears2pecan(geofile, outdir, start, end, product, projection=None):
+    """
+    Downloads remote sensing data from AppEEARS
+
+    Parameters
+    ----------
+    geofile (str) -- path to the GeoJSON file containing the name and coordinates of AOI
+    
+    outdir (str) -- path to the directory where the output file is stored. If specified directory does not exists, it is created.
+  
+    start (str) -- starting date of the data request in the form YYYY-MM-DD
+    
+    end (str) -- ending date area of the data request in the form YYYY-MM-DD
+
+    product (str) -- product name followed by " . " and the product version, e.g. "SPL3SMP_E.003"
+
+    projection (str) -- type of projection, only required for polygon AOI type. None by default
+
+    Returns
+    -------
+    Nothing:
+            Output files are saved in the specified directory.
+            Output file is of netCDF type when AOI is a Polygon and csv type when AOI is a Point.
+    """
+
+    # API url
+    api = "https://lpdaacsvc.cr.usgs.gov/appeears/api/"
+
+    def authenticate():
+        """
+        uses the user provided NASA Earthdata credentials to request a token.
+        
+        Returns
+        -------
+        head (dict) : header contatining the authentication token 
+        """
+        try:
+            # if the user does not want to enter their credentials everytime they use this function, they need to store their username and password in a JSON file, preferabbly not in a git initialized directory
+            with open("path/to/cred.json", "r") as f:
```

Comment:
Yes, currently that needs to be done. Passing it as an argument is an option. If the user knows where to store the file then maybe I can also create the JSON file automatically once the user enters their credentials for the first time and then use that for subsequent runs.

Reply:
Yeah I think any solution that doesn't require touching code would be better. Otherwise, this also makes other kinds of small nuisance possible. e.g. others to commit their hardcoded paths by mistake etc.